proposals/lip-0051.md

```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
 
 This function returns token IDs for both native and non-native tokens supported by the chain.
 
+```python
+def getSupportedTokens() -> dict[list]:
+    # Native tokens
+    nativeTokenIDs = [tokenID for tokenID in supplyStore.keys()]
+
+    # Non-native tokens	
+    if supportedTokens(ALL_SUPPORTED_TOKENS_KEY) exists:
+        return {
+            "nativeTokenIDs": nativeTokenIDs,
+            "supportedChainIDs": "All",
+            "supportedTokenIDs": "All"
+        }
+
+    supportedChainIDs = []
+    supportedTokenIDs = [getMainchainID()]
+    
+    for chainID in supportedTokens.keys():
+        if supportedTokens(chainID).supportedTokenIDs == []:
+            supportedChainIDs.append(chainID)
```

Comment:
In order to be more aligned with the SDK I would write here `supportedTokenIDs.append(chainID+"********")`.